helm/templates/03-deployment.yaml

```
@@ -140,4 +140,27 @@ spec:
           persistentVolumeClaim:
             claimName: {{ template "appname" . }}-pvc
       {{- end }}
+      {{ if or (.Values.replicaCount > 1) .Values.hpa.enabled }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  name: app
```

Correct my if I'm wrong but wouldn't this label selector match on pods that have a label `name=app`.

I think that what we actually want is more like:
```suggestion
                  app: {{ template "appname". }}
```


We could add the two other labels, `release` and `track`, as well.
```suggestion
                  app: "{{ template "appname". }}"
                  release: "{{ .Release.Name }}"
                  track: "{{ .Values.Application.track }}"
```